src/math_utils.rs

```diff
@@ -82,17 +102,25 @@ pub fn ec_add(
     point_b: (BigInt, BigInt),
     prime: &BigInt,
 ) -> (BigInt, BigInt) {
-    let m = line_slope(point_a.clone(), point_b.clone(), prime);
+    let m = line_slope(&point_a, &point_b, prime);
     let x = (m.clone() * m.clone() - point_a.0.clone() - point_b.0).mod_floor(prime);
     let y = (m * (point_a.0 - x.clone()) - point_a.1).mod_floor(prime);
     (x, y)
 }
 
 /// Computes the slope of the line connecting the two given EC points over the field GF(p).
 /// Assumes the points are given in affine form (x, y) and have different x coordinates.
-pub fn line_slope(point_a: (BigInt, BigInt), point_b: (BigInt, BigInt), prime: &BigInt) -> BigInt {
-    assert!((point_a.0.clone() - point_b.0.clone()) % prime != bigint!(0));
-    div_mod(point_a.1 - point_b.1, point_a.0 - point_b.0, prime.clone())
+pub fn line_slope(
+    point_a: &(BigInt, BigInt),
+    point_b: &(BigInt, BigInt),
+    prime: &BigInt,
+) -> BigInt {
+    assert!((point_a.0.clone() - point_b.0.clone()) % prime != bigint!(0_i32));
```

Comment:
Why the clones? Also, let's us `is_zero`.

Reply:
resolved:
- https://github.com/lambdaclass/cleopatra_cairo/commit/a0008d67fc3cf60bc34e3cd8d19773e833d52772
- https://github.com/lambdaclass/cleopatra_cairo/commit/83bde4c8487d60e10b3bd34bc4e9b7cb23857f5c